### PR TITLE
Allow dma capable to be permissive

### PR DESCRIPTION
**Pull request by davids5**
## Summary
  A driver passing a buffer that is aligned and sized on d-cache bounders but does not used the full extent of the buffer will fail the dma_capable test and in a debug build DEBUGASSERT. 

This PR adds the option configures the stm32_dmacapable to not disqualify DMA operations on memory that is not dcache aligned based solely on the starting addresss and byte count.

One would use this when ALL buffer extents are known to be aligned, but the the count does not use the complete buffer.

## Impact
  None - if disabled and it is by default.

## Testing

Run a debug build and pass a buffer that is 64 bytes and aligned on 32, but a count of 16 bytes to DMA. It will no longer assert.

### CODE REVIEW

**Comment by davids5** [comment]
Please ignore (or suggest a remedy) for the alignment of the braces. It looks fine to me, but the ifdef must be screwing with the detection.

File: arch/arm/src/stm32h7/Kconfig [hunk 1]
```diff
@@ -1684,6 +1684,16 @@ config STM32H7_DMACAPABLE
...
 
+config STM32H7_DMACAPABLE_ASSUME_CACHE_ALIGNED
+	bool "Do not disqualify DMA capability based on cache alignment"
+	depends on STM32H7_DMACAPABLE && ARMV7M_DCACHE && !ARMV7M_DCACHE_WRITETHROUGH
+	default n
+	---help---
+		This option configures the stm32_dmacapable to not disqualify
+		DMA operations on memory that is not dcache aligned based solely
+		on the starting addresss and byte count.
+		Use this when ALL buffer extents are known to be alligned, but the
```
Comment (ghn-certi) [inline arch/arm/src/stm32h7/Kconfig:1695]:
```suggestion
		Use this when ALL buffer extents are known to be aligned, but the
```

**Review by xiaoxiang781216 — COMMENTED** [review]
LGTM, let's address @ghn-certi's comment.

**Comment by jerpelea** [comment]
LGTM please fix CI error

File: arch/arm/src/stm32h7/Kconfig (hunk 1 above)
Comment (davids5) [reply arch/arm/src/stm32h7/Kconfig]: @ghn-certi - Thank you for the review. Fixed and force pushed.

**Comment by davids5** [comment]
> LGTM please fix CI error

As I said above: please ignore (or suggest a remedy) for the alignment of the braces. It looks fine to me, but the `ifdef` must be screwing with the detection.

Hi @jerpelea - any suggestions on how to fix it?

**Comment by jerpelea** [comment]
we have 2 options 
- add 4 spaces instead of 2 before all brackets and we avoid future errors
- ignore the error

**Comment by Ouss4** [comment]
> @davids5 we have 2 options
> 
>     * add 4 spaces instead of 2 before all brackets and we avoid future errors
> 
>     * ignore the error

I'd suggest we ignore it for this PR and address the nxstyle issue later.  The alignment seems fine and we have a lot of similar cases in the code base.

**Review by Ouss4 — APPROVED** [review]

**Comment by jerpelea** [comment]
> 
> 
> > @davids5 we have 2 options
> > ```
> > * add 4 spaces instead of 2 before all brackets and we avoid future errors
> > 
> > * ignore the error
> > ```
> 
> I'd suggest we ignore it for this PR and address the nxstyle issue later. The alignment seems fine and we have a lot of similar cases in the code base.

+1

**Comment by davids5** [comment]
@jerpelea @Ouss4 - Agreed! - Who would like to push the merge button?

**Pull request merged**